morecantile/models.py

```diff
@@ -765,9 +770,8 @@ def custom(
 
     def matrix(self, zoom: int) -> TileMatrix:
         """Return the TileMatrix for a specific zoom."""
-        for m in self.tileMatrices:
-            if m.id == str(zoom):
-                return m
+        if (idx := self._tile_matrices_idx.get(str(zoom), None)) is not None:
```

Comment:
hmm why not just have the keys be ints within `self._tile_matrices_idx`?  Also a lru_cache for the function might also be interesting to consider